src/app/clusters/ota-provider/ota-provider.cpp

```diff
@@ -55,7 +55,7 @@ constexpr size_t kMaxMetadataLen       = 512; // The maximum length of Metadata
 constexpr size_t kUpdateTokenMaxLength = 32;  // The expected length of the Update Token parameter used in multiple commands
 constexpr size_t kUpdateTokenMinLength = 8;   // The expected length of the Update Token parameter used in multiple commands
 
-OTAProviderDelegate * gDelegateTable[kOtaProviderDelegateTableSize] = { nullptr };
+OTAProviderDelegate * gDelegateTable[kOtaProviderDelegateTableSize];
```

Comment:
So there might be another issue here: if kOtaProviderDelegateTableSize is 0, this is not actually valid C++ and compilers are allowed to reject it.

Given that, maybe we should have a compile-time define for "this thing is nonzero" and condition the definition on that, defining `OTAProviderDelegate ** gDelegateTable = nullptr` when the count is 0?  And then we might not need an init at all..

Reply:
yes, it works, thanks